src/core/async-work-cache.ts

```diff
@@ -31,11 +35,34 @@ export class AsyncWorkCache<K, V> {
    *
    * If not, compute it with the given function.
    *
-   * This method ensures that we will only try to compute the value for `key`
-   * once, no matter how often or with what timing getOrCompute is called, even
-   * recursively.
+   * This method ensures that, in the absence of cancellations, we will only try
+   * to compute the value for `key` once, no matter how often or with what
+   * timing getOrCompute is called, even recursively.
+   *
+   * If a compute() throws a Cancel that should not result in a Cancel for
```

Comment:
I don't quite understand the first sentence here.

Reply:
Rephrased, PTAL